answer/lib.rs

```diff
@@ -81,62 +81,62 @@ impl Type {
         }
     }
 
-    pub fn as_object_type(&self) -> ObjectType<'static> {
+    pub fn as_object_type(&self) -> ObjectType {
         match self {
-            Type::Entity(entity) => entity.clone().into_owned_object_type(),
-            Type::Relation(relation) => relation.clone().into_owned_object_type(),
+            Type::Entity(entity) => (*entity).into_owned_object_type(),
+            Type::Relation(relation) => (*relation).into_owned_object_type(),
             Type::Attribute(_) => panic!("Attribute Type is not an object type"),
             Type::RoleType(_) => panic!("Role Type is not an Object type"),
         }
     }
 
-    pub fn as_entity_type(&self) -> EntityType<'static> {
+    pub fn as_entity_type(&self) -> EntityType {
         match self {
-            Type::Entity(entity) => entity.clone().into_owned(),
+            Type::Entity(entity) => (*entity).into_owned(),
```

Comment:
Some of these `into_owned()` calls are no-ops. I've cleaned up a few, but there are simply too many.